### ISSUE TITLE
Connect: Add a way to manually relogin to a cluster

### ISSUE THREAD

**Issue by ravicious** [gravitational/teleport#21385]
### What would you like Teleport to do?

There should be a way in Connect to do the equivalent of `tsh login --proxy=<current profile proxy>` without logging out first.

### What problem does this solve?

Teleport stores crucial information in the cert created during login. Admins of the cluster might make changes to a particular account, such as adding a new role, which won't be reflected until the user creates a new cert. Being able to refresh this cert at will feels like a crucial part of the workflow.

### If a workaround exists, please include it.

At the moment, the user can log out of the cluster and log in again. This has a number of disadvantages:

* The user needs to retype the cluster address and the credentials if SSO is not used.
* Logging out wipes all cluster-specific state from Connect. This includes the list of recently used connections and along with it port numbers for previously set up database connections.

We might want to reconsider if wiping this state on logout is necessary. AFAIR it was there only because we wanted to prevent the app from accumulating cruft from clusters the users no longer were using (@gzdunek do you remember anything more?). We could replace that with a more sophisticated solution: let's say we'd wipe the cluster state only if X amount of time passed after the user logged out of the cluster by using the logout button.

**Comment by gzdunek** [comment]
I think the main reason was not to confuse users by showing them connections from clusters they no longer have.

I'm not sure about the proposed changes to state wiping. Won't we have to deal with edge cases like closing the app before the time passes? It also doesn't solve the first point (retyping the cluster address and credentials).

Maybe we could rework the user profile dropdown? Instead of having only logout button, we would make each item expandable, with the options:
- Re-login
- Logout

Something like here:
![image](https://user-images.githubusercontent.com/20583051/217227798-0812c0f3-17b1-44e5-8181-0a5154a44b01.png)

OTOH I can imagine that most of the users doesn't need it, so this additional option can be confusing for them.

Or we can just add a 'Re-login' button at the top of the dropdown.

**Comment by ravicious** [comment]
> I think the main reason was not to confuse users by showing them connections from clusters they no longer have.

Yes, but we could hide connections from clusters the user has manually logged out of.

> Won't we have to deal with edge cases like closing the app before the time passes?

We could do this on login or whatever, I don't understand what this edge case is about but I suppose it assumes a certain way of implementing what I suggested?

> It also doesn't solve the first point (retyping the cluster address and credentials).

My assumption was that if the user manually logs out, we keep the workspace state but don't show it in the regular profile selector. Then once they attempt to add a new cluster, we could suggest the address of one of previously used clusters ✨somehow✨.

**Comment by ravicious** [comment]
See [the Slack thread](https://gravitational.slack.com/archives/C03FJA391M3/p1677859079388879) for possible solutions.

**Comment by webvictim** [comment]
Related: https://github.com/gravitational/teleport/issues/28368

**Comment by ravicious** [comment]
This was added in #51419.
